lib/collection.dart

```diff
@@ -77,3 +77,14 @@ bool setsEqual(Set a, Set b) {
 
   return a.containsAll(b);
 }
+
+/// Returns the index of the first item in [list] where [matcher] evaluates to
+/// true.
+///
+/// Returns -1 if there are no items where [matcher] evaluates to true.
+int indexOf<T>(List<T> list, bool matcher(T item)) {
```

Comment:
I'd imagine this utility could be useful for any `Iterable`. Are we worried about performance? If so, we could do:

```dart
int indexOf<T>(Iterable<T> elements, bool predicate(T element)) {
  if (elements is List) {
    for (int i = 0; i < element.length; i++) {
      ...
    }
    return -1;
  }

  for (T element in elements) {
    ...
  }
  return -1;
}
```